src/main/java/com/pusher/client/Pusher.java

```diff
@@ -352,10 +361,32 @@ public PresenceChannel subscribePresence(final String channelName, final Presenc
         final PresenceChannelImpl channel = factory.newPresenceChannel(connection, channelName,
                 pusherOptions.getAuthorizer());
         channelManager.subscribeTo(channel, listener, eventNames);
-
+        presenceChannels.add(channel);
         return channel;
     }
 
+    public Channel getChannel(String channelName){
+        return getChannelFromList(channelName, channels);
+    }
+
+    public PrivateChannel getPrivateChannel(String channelName){
+        return getChannelFromList(channelName, privateChannels);
+    }
+
+    public PresenceChannel getPresenceChannel(String channelName){
```

Comment:
this should only be working if the channel name starts with "presence-", otherwise throw an exception, similar for private and public channels